internal/pkg/telemetry/span.go

```diff
@@ -1,19 +1,33 @@
 package telemetry
 
 import (
+	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
 )
 
-func EndSpan(span trace.Span, errPtr *error) {
+type Span interface {
+	End(errPtr *error, opts ...trace.SpanEndOption)
+	SetAttributes(kv ...attribute.KeyValue)
+}
```

Comment:
OTEL span ma kopec dalsich method, ktore nechcem `SetName(), ....`.
https://pkg.go.dev/go.opentelemetry.io/otel/trace#Span

Ciel je, aby to bolo vsade rovnako a dalo sa to pouzit iba jednym sposobom.